arch/arm64/asm.S

```diff
@@ -83,5 +83,60 @@ FUNCTION(arm64_el3_to_el1)
 
     eret
 
+FUNCTION(arm64_elX_to_el1)
```

Comment:
Should we either save off x0 or use something >x3 in these routines since it is possible that arguments could be passed to the kernel in x0-x3?  Based on where these routines are likely to be used we could be mangling args passed to kernel.